### DISCUSSION TITLE
Does not export posts per subtree

### DISCUSSION THREAD

**Discussion by louis-vgn** [kaushalmodi/ox-hugo#751] [General]
I’m trying to publish everything with the preferred workflow (subtree) posts, but I can’t figure out why it’s not working. 

I followed the documentation, instructions, and examples for real world websites, yet it still doesn’t work. Instead of exporting all subtrees to  multiple markdown files in their respecting directories, it only exports he posts, not the pages, and the second post is not under the `articles` bundle. 

Here is the file I am trying to export. I really don’t know what I am doing wrong. Any help would be very welcome. 

```org
:CONFIG:
#+HUGO_BASE_DIR: ../
#+HUGO_AUTO_SET_LASTMOD: t
#+OPTIONS: creator:t
:END:

#+title: source file
#+author: author
#+email: example@tld


* Pages

This section contains all the static pages

** DONE About / À propos
CLOSED: [2024-12-21 Sat 20:19]
:LOGBOOK:
- State "DONE"       from "TODO"       [2024-12-21 Sat 20:19]
:END:
:PROPERTIES:
:EXPORT_HUGO_SECTION: /
:EXPORT_FILE_NAME: about
:EXPORT_HUGO_MENU: :menu main :weight 60
:END:

this is about 


** DONE Resume / CV
CLOSED: [2024-12-21 Sat 20:19]
:LOGBOOK:
- State "DONE"       from "TODO"       [2024-12-21 Sat 20:19]
:END:
:PROPERTIES:
:EXPORT_HUGO_SECTION: /
:EXPORT_FILE_NAME: resume
:EXPORT_HUGO_MENU: :menu main :weight 50
:END:


this is my resume

* Posts
:PROPERTIES:
:EXPORT_HUGO_BUNDLE: posts
:EXPORT_FILE_NAME: _index
:EXPORT_HUGO_MENU: :menu posts

** Tech
:PROPERTIES:
:EXPORT_HUGO_BUNDLE: tech
:EXPORT_HUGO_FILE_NAME: _index
:EXPORT_HUGO_MENU: :menu tech :parent posts
:END:

*** TODO Test post
:PROPERTIES:
:EXPORT_FILE_NAME: test-post
:END:
:LOGBOOK:
- State "DONE"       from              [2024-12-22 Sun 18:17]
:END:

Test posts


** articles
:PROPERTIES:
:EXPORT_HUGO_SECTION: articles
:EXPORT_HUGO_BUNDLE: articles
:EXPORT_FILE_NAME: _index
:EXPORT_HUGO_MENU: :menu articles

*** DONE test post
CLOSED: [2024-12-22 Sun 18:23]
:PROPERTIES:
:EXPORT_FILE_NAME: test-post-2
:END:
:LOGBOOK:
- State "DONE"       from              [2024-12-22 Sun 18:23]
- State "DONE"       from              [2024-12-22 Sun 18:17]
:END:

Test posts
```

The end results is the following (`tree content/`):

```
.
└── posts
    ├── tech
    │   └── test-post.md
    └── test-post-2.md
```

**Comment by louis-vgn** [comment]
Ok maybe I can narrow that down.

Pages section are not being exported.  I get the error "Point is not a valid Hugo post subtree; move to one and try again".

So the question is what is a valid Hugo post subtree?

In the doc I can find the following: 

> A valid Hugo post subtree is an Org subtree that has the EXPORT_FILE_NAME property set. Note that a subtree with EXPORT_FILE_NAME property cannot nest a subtree with the same property set. If you can analogize with the [branch/leaf data structure terminlogy](https://en.wikipedia.org/wiki/Tree_(data_structure)), then the subtrees with EXPORT_FILE_NAME property need to be leaf nodes.

I do have the file name, so it should be exported? Or maybe that’s not considered a post, and needs something else?

**Comment by louis-vgn** [comment]
Ok maybe I can narrow that down.

Pages section are not being exported.  I get the error "Point is not a valid Hugo post subtree; move to one and try again".

So the question is what is a valid Hugo post subtree?

In the doc I can find the following: 

> A valid Hugo post subtree is an Org subtree that has the EXPORT_FILE_NAME property set. Note that a subtree with EXPORT_FILE_NAME property cannot nest a subtree with the same property set. If you can analogize with the [branch/leaf data structure terminlogy](https://en.wikipedia.org/wiki/Tree_(data_structure)), then the subtrees with EXPORT_FILE_NAME property need to be leaf nodes.

I do have the file name, so it should be exported? Or maybe that’s not considered a post, and needs something else?

**Comment by kaushalmodi** [comment]
Hi Louis,

I'm travelling and on phone, so cannot run your example. But I see 2 main issues:

1. You are nesting nodes with `:EXPORT_FILE_NAME:` properties. E.g. if a `*` level mode has that property, you cannot have the same property again at `**`, `***` or any of the deeper levels. 
2. I also noticed a `:EXPORT_HUGO_FILE_NAME:` property which is not recognized. May be that was a typo?

**Comment by kaushalmodi** [comment]
Check out this example from the test suite on how you can organize the page bundles: https://github.com/kaushalmodi/ox-hugo/blob/main/test/site/content-org/deep-nesting.org?plain=1

**Comment by louis-vgn** [comment]
First of, thanks for your time and response!

Indeed, `:EXPORT_HUGO_FILE_NAME:` was a typo. However, even when trying to replicate the example given, and removing the nesting of `:EXPORT_FILE_NAME:` nodes, I still get the error `this is not a valid Hugo post subtree`…

Just to be clear, copying and pasting the example file exports properly. It’s just me trying to replicate that does not export. Clearly something’s wrong with me. I can’t figure it out! 

```
* Pages

This section contains all the static pages

** Homepage
:PROPERTIES:
:EXPORT_TITLE: Personal blog
:EXPORT_FILE_NAME: _index
:EXPORT_HUGO_TYPE: homepage
:EXPORT_HUGO_SECTION: /
:EXPORT_HUGO_MENU: :menu "1.main" :title "Homepage"
:CUSTOM_ID: main
:END:

This is my homepage

** About
:PROPERTIES:
:EXPORT_HUGO_SECTION: about
:END:

*** DONE my About
CLOSED: [2024-12-21 Sat 20:19]
:LOGBOOK:
- State "DONE"       from "TODO"       [2024-12-21 Sat 20:19]
:END:
:PROPERTIES:
:EXPORT_FILE_NAME: my_about
:END:

this is about 


*** DONE Resume / CV
CLOSED: [2024-12-21 Sat 20:19]
:LOGBOOK:
- State "DONE"       from "TODO"       [2024-12-21 Sat 20:19]
:END:
:PROPERTIES:
:EXPORT_FILE_NAME: resume
:END:


this is my resume
```

That’s what I have. The homepage works fine. However, both the resume section and the about section do not export with the aforementioned error. 

I think I’m mainly confused with the `section` and `page bundle`. In the example file, it’s using sections, but I think I should be using a leaf page bundle?

**Reply by louis-vgn** [reply]
Ok, removing the `DONE` todo keyword, and now everything exports fine!

I thought that ox-hugo would look in the custom variables of org-mode to see the special property `CLOSED`.

**Reply by kaushalmodi** [reply]
DONE is indeed supported. I'll get a chance to try your example after a week.

**Reply by louis-vgn** [reply]
For some reason, I have changed a few things randomly (as I was pulling my hair out), and now it works. I’m not sure what’s not working with the example provided, but for now my setup seems to work.

**Comment by Nika250** [comment]
Interesting, thank you!
